drivers/dp/swdp_bitbang.c

```diff
@@ -101,124 +103,108 @@ static ALWAYS_INLINE uint32_t sw_get32bit_parity(uint32_t data)
 	return data & 1U;
 }
 
-static ALWAYS_INLINE void pin_delay_asm(uint32_t delay)
-{
-#if defined(CONFIG_SOC_SERIES_NRF52X) || defined(CONFIG_SOC_FAMILY_LPC)
-	__asm volatile ("movs r3, %[p]\n"
-			".start_%=:\n"
-			"subs r3, #1\n"
-			"bne .start_%=\n"
-			:
-			: [p] "r" (delay)
-			: "r3", "cc"
-			);
-#else
-#error "Not defined for this SoC family"
-#endif
-}
-
-static ALWAYS_INLINE void pin_platform_set(void *base, uint32_t port, uint8_t pin)
-{
-#if defined(CONFIG_SOC_SERIES_NRF52X)
-	NRF_GPIO_Type * reg = base;
-
-	reg->OUTSET = BIT(pin);
-#elif defined(CONFIG_SOC_FAMILY_LPC)
-	GPIO_Type *gpio_base = base;
-
-	gpio_base->SET[port] = BIT(pin);
-#else
-#error "Not defined for this SoC family"
-#endif
-}
-
-static ALWAYS_INLINE void pin_platform_clr(void *base, uint32_t port, uint8_t pin)
-{
-#if defined(CONFIG_SOC_SERIES_NRF52X)
-	NRF_GPIO_Type * reg = base;
-
-	reg->OUTCLR = BIT(pin);
-#elif defined(CONFIG_SOC_FAMILY_LPC)
-	GPIO_Type *gpio_base = base;
-
-	gpio_base->CLR[port] = BIT(pin);
-#else
-#error "Not defined for this SoC family"
-#endif
-}
-
-static ALWAYS_INLINE uint32_t pin_platform_get(void *base, uint32_t port, uint8_t pin)
-{
-#if defined(CONFIG_SOC_SERIES_NRF52X)
-	NRF_GPIO_Type * reg = base;
-
-	return ((reg->IN >> pin) & 1);
-#elif defined(CONFIG_SOC_FAMILY_LPC)
-	GPIO_Type *gpio_base = base;
-
-	return ((gpio_base->PIN[port] >> pin) & 1);
-#else
-#error "Not defined for this SoC family"
-#endif
-}
-
 /* Set SWCLK DAP hardware output pin to high level */
 static ALWAYS_INLINE void pin_swclk_set(const struct device *dev)
 {
 	const struct sw_config *config = dev->config;
-	const struct gpio_dt_spec *dt_spec = &config->clk;
 
-	pin_platform_set(config->clk_reg, config->clk_port, dt_spec->pin);
+#if FAST_BITBANG_HW_SUPPORT
+	pin_platform_set(config->clk_reg, config->clk_port, config->clk.pin);
+#else
+	gpio_pin_set_dt(&config->clk, 1);
+#endif /* FAST_BITBANG_HW_SUPPORT */
```

Comment:
Please not more ifdefery.
```c
	if (FAST_BITBANG_HW_SUPPORT) {
		swdp_ll_pin_set(config->clk_reg, config->clk.pin);
	} else {
		gpio_pin_set_dt(&config->clk, 1);
	}
```

Reply:
The main thing is that if you do `if (FAST_BITBANG_HW_SUPPORT) {` as proposed by @jfischer-no instead of `#if FAST_BITBANG_HW_SUPPORT` then the code will fail to compile if there is error regardless of the `FAST_BITBANG_HW_SUPPORT` value. With ifdef, the path that is not in the default configuration is likely to get completely broken over time when the other parts of code change, but the inactive ifdef is left intact.

Reply:
done